src/display/annotation_layer.js

```diff
@@ -198,7 +198,25 @@ class AnnotationElement {
       page.view[3] - data.rect[3] + page.view[1],
     ]);
 
-    container.style.transform = `matrix(${viewport.transform.join(",")})`;
+    if (data.hasOwnCanvas) {
+      const transform = viewport.transform.slice();
+      const scale = Math.abs(transform[0] || transform[1]);
+      width = Math.ceil(width * scale);
```

Comment:
Does it make sense here to use `Util.singularValueDecompose2dScale` so we can handle different scaling in x/y?

Reply:
Is it possible to have a different scaling for x/y ?